### PR TITLE
only pin aiohttp downwards for py3.11 compat

### PR DESCRIPTION
**Pull request by janosh**
Running

```sh
pip install covalent
```

fails on Python 3.11:

```
      copying aiohttp/.hash/hdrs.py.hash -> build/lib.macosx-10.9-universal2-cpython-311/aiohttp/.hash
      running build_ext
      building 'aiohttp._websocket' extension
      creating build/temp.macosx-10.9-universal2-cpython-311
      creating build/temp.macosx-10.9-universal2-cpython-311/aiohttp
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/Users/janosh/.venv/py311/include -I/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c aiohttp/_websocket.c -o build/temp.macosx-10.9-universal2-cpython-311/aiohttp/_websocket.o
      aiohttp/_websocket.c:198:12: fatal error: 'longintrepr.h' file not found
        #include "longintrepr.h"
                 ^~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for aiohttp
Failed to build aiohttp
ERROR: Could not build wheels for aiohttp, which is required to install pyproject.toml-based projects
```

Looks like [`aiohttp==3.8.2`](https://github.com/aio-libs/aiohttp/issues/6600) fixed install on 3.11 so removing the strict pin should fix this.

### CODE REVIEW

**Comment by janosh** [comment]
Btw, unless it solves a specific issue, I would generally advise against strict dep pins.

**Comment by santoshkumarradha** [comment]
@janosh thanks for this ! Officially we have only QA'd covalent from 3.8-3.10, although this won't affect this PR, while the team looks at it, did the pytest pass in this PR with Python 3.11? If so that might be nice to know. 

CC: @AlejandroEsquivel

**Comment by janosh** [comment]
> did the pytest pass in this PR with Python 3.11?

They did not. But that's because I didn't complete the setup. The errors didn't sound Python version specific:

<details><summary>Details</summary>

```py
FAILED covalent_ui/api/v1/tests/overview_test.py::test_overview - sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: l...
FAILED tests/covalent_dispatcher_tests/_cli/migrate_test.py::test_process_legacy_node - TypeError: code() argument 13 must be str, not int
FAILED tests/covalent_dispatcher_tests/_cli/migrate_test.py::test_process_transport_graph - TypeError: code() argument 13 must be str, not int
FAILED tests/covalent_dispatcher_tests/_cli/migrate_test.py::test_process_transport_graph_is_idempotent - TypeError: code() argument 13 must be str, not int
FAILED tests/covalent_dispatcher_tests/_cli/migrate_test.py::test_process_lattice - TypeError: code() argument 13 must be str, not int
FAILED tests/covalent_dispatcher_tests/_cli/migrate_test.py::test_process_result_object - TypeError: code() argument 13 must be str, not int
FAILED tests/functional_tests/basic_dispatcher_test.py::test_dispatcher_functional - ConnectionError: The Covalent dispatcher server is not running at http://loc...
FAILED tests/functional_tests/choose_executor_test.py::test_using_executor_names - ConnectionError: The Covalent dispatcher server is not running at http://loc...
FAILED tests/functional_tests/choose_executor_test.py::test_using_executor_classes - ConnectionError: The Covalent dispatcher server is not running at http://loc...
```
</details>

**Comment by arosen93** [comment]
My guess is the version check failed because a released version's notes were updated (even though it was just a formatting change).

File: CHANGELOG.md [hunk 1]
```diff
@@ -289,7 +290,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unit and functional tests for redispatching.
 
 ### Docs
-- Updated self-deployment (server deployment). 
+- Updated self-deployment (server deployment).
```
Comment (AlejandroEsquivel) [inline CHANGELOG.md:293]:
```suggestion
- Updated self-deployment (server deployment).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2010,7 +2011,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Updated `covalent db migrations` to overwrite `alembic.ini` `script_location` with absolute path to migrations folder
-- Updated `covalent db alembic [args]` command to use project root as `cwd` for alembic subprocess  
+- Updated `covalent db alembic [args]` command to use project root as `cwd` for alembic subprocess
```
Comment (AlejandroEsquivel) [inline CHANGELOG.md:2014]:
```suggestion
- Updated `covalent db alembic [args]` command to use project root as `cwd` for alembic subprocess  
```

File: CHANGELOG.md (hunk 1 above)
Comment (AlejandroEsquivel) [inline CHANGELOG.md:293]:
```suggestion
- Updated self-deployment (server deployment). 
```

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]
Appreciate the contribution @janosh just fixing some minor spacing on changelog as our changelog action is quite picky about changes outside the [UNRELEASED] block

**Review by AlejandroEsquivel — APPROVED** [review]

**Pull request merged**